### PR TITLE
Use consistent new og image style design and custom metadata across pages 

### PR DESCRIPTION
**Pull request by sheriffjimoh**
## Summary by CodeRabbit

- **New Features**
  - Introduced a new endpoint for generating Open Graph images based on thread content.
  - Added a component to generate SVG background images based on light or dark themes.
  - Added a component to generate styled images for threads.

- **Enhancements**
  - Improved metadata generation for threads, including title, publish date, summary, and social media details.
  - Enhanced thread operations with new functions for message processing and link generation.

- **Dependencies**
  - Updated dependencies to include `"zod": "^3.23.8"` and `"remark-math": "^5.1.1"`.

- **Bug Fixes**
  - Updated `GetThreadParams` interface to allow `threadId` to be `string` or `null`.

### CODE REVIEW

File: apps/masterbots.ai/app/og/route.tsx [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+      thread.messages.find(m => m.role === 'assistant')?.content || 'not found'
+    const username = thread.user?.username
+    const user_avatar = thread.user?.profilePicture || ''
+
+    let theme = 'light'
+    if (typeof window !== 'undefined') {
+      theme = localStorage.getItem('theme') || 'dark'
+    }
+    const isLightTheme = theme === 'light'
+    return new ImageResponse(
```
Comment (AndlerRL) [inline apps/masterbots.ai/app/og/route.tsx:28]: Let's improve on code split here. Please create components for:

- The SVG (as background image)
- Image Response

File: apps/masterbots.ai/lib/metadata.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   const firstQuestion=
+      thread.messages.find(m => m.role === 'user')?.content || 'not found'
+    const firstResponse =
+      thread.messages.find(m => m.role === 'assistant')?.content || 'not found'
+
+  const data = {
+    title: firstQuestion,
+    publishedAt: thread.updatedAt, // format(thread.updatedAt, 'MMMM dd, yyyy'),
+    summary: firstResponse,
+    image: `http://localhost:3000/og?threadId=${thread.threadId}`,
```
Comment (AndlerRL) [inline apps/masterbots.ai/lib/metadata.ts:23]: Change base url to the `process.env.VERCEL_URL` variable and do the same for the object below on the returning block.

File: apps/masterbots.ai/lib/url.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { z, ZodSchema } from 'zod'
+
+// Zod schema for validating slug strings
+export const SlugSchema: ZodSchema<string> = z
+  .string()
+  .min(1)
+  .regex(/^[a-z0-9]+[a-z0-9+_-]*[a-z0-9]+$/, 'Invalid slug format.')
+
+// Function to convert a username into a slug
+export const toSlug = (username: string, separator = '_'): string => {
```
Comment (AndlerRL) [inline apps/masterbots.ai/lib/url.ts:10]: Make sure to spread this functions everywhere. There is already a `toSlug` function on the same folder. Check this: https://github.com/bitcashorg/masterbots/blob/f91a739a131799a9e931ee6052d61801a73208ee/apps/masterbots.ai/lib/utils.ts#L185

Replace the regExp ereplacement from `_and_` to `_n_`. It is shorter and familiar for users.

**Review by AndlerRL — CHANGES_REQUESTED** [review]
Changes LGTM, I noticed some improvement that it can be done before to merge. I will be testing also after those observations are done.

File: apps/masterbots.ai/lib/metadata.ts (hunk 2 above)
Comment (AndlerRL) [reply apps/masterbots.ai/lib/metadata.ts:23]: image
 the preview one can be changed to the preview URL given on this PR.

**Review by AndlerRL — APPROVED** [review]

**Comment by AndlerRL** [comment]
Hey @sheriffjimoh , do you see your IDE with these highlights? (tab text colour and the line 12 on the code.

image

I noticed that the build failed to compile, see: https://vercel.com/bitcash/masterbots/EPik13nfAqvLtaAXiKm5QjSJVQYN and there is more than 1 issue on that file. Can you please check it out. This is ready to merge (based on the task) but I can´t if it is not compiling.

**Comment by gaboesquivel** [comment]
@sheriffjimoh @AndlerRL make sure all these changes are applied on search branch too

**Comment by AndlerRL** [comment]
[Create Dynamic OG Image](https://trello.com/c/IC71JYlv/153-create-dynamic-og-image)

**Review by AndlerRL — APPROVED** [review]
@sheriffjimoh — I will be merging this, all LGTM. Remember to raise the PR to the `search` branch. 👍

**Pull request merged**